### ISSUE TITLE
[BUG] Children Grid Template button is dimmed

### ISSUE THREAD

**Issue by benshen-secretlab** [pimcore/studio-backend-bundle#1863]
### Pimcore version
v2026.01.02

### Steps to reproduce
Issue 1:
1. Go to any data object that has children skus and click on "children grid"
2. In the right side panel, click on search and filer. Add some filters
3. In the same panel, click on grid config. Reorganise / remove some columns.
4. Click on "Save as template"

<img width="490" height="268" alt="Image" src="https://github.com/user-attachments/assets/773d2b94-fec9-4321-a301-c1d5f009466a" />

Upon checked from database `bundle_studio_grid_configurations`, all the submitted value were saved. However the button from front-end is dimmed.

<img width="1274" height="512" alt="Image" src="https://github.com/user-attachments/assets/77555e12-658e-4f40-920f-fcf20f8cb2b8" />

Issue 2:
The cancel button is not responding for enlarge grid config modal

<img width="2254" height="1216" alt="Image" src="https://github.com/user-attachments/assets/0ad58f2f-9171-4535-9ab1-d82c01607726" />

### Actual Behavior
Issue 1: Template button is dimmed.
Issue 2: Cancel button not responding.

### Expected Behavior
Issue 1: Template button should be able to click and list down all the saved templates.
Issue 2: Cancel button should close the modal

**Comment by benshen-secretlab** [comment]
Network show the endpoint get fired to retrieve the list of saved template. However, frontend button is not responding with the listing from ajax.

<img width="1182" height="628" alt="Image" src="https://github.com/user-attachments/assets/efa9568b-cc50-4a20-b7a6-31e65c6e7258" />

**Comment by benshen-secretlab** [comment]
would anyone be able to look into this issue?

**Comment by ValeriaMaltseva** [comment]
@benshen-secretlab
Thanks for raising the issue and following up :)
The issue is on our radar and we will take a closer look as part of the current priorities and workload. Yesterday was a public holiday, so responses have been a bit slower than usual.

**Comment by benshen-secretlab** [comment]
Hi @ValeriaMaltseva 

We are currently using the Grid Config template feature to manage different views for different roles. In the meantime, do you have any alternative suggestions from Pimcore Studio that allows us to handling this kind of view management?

**Comment by markus-moser** [comment]
@benshen-secretlab We will check it in detail soon. But until then, I quickly checked it. When I add a grid template directly to the folder where the items are contained (or directly to the children grid) it's listed normally:

<img width="1912" height="1336" alt="Image" src="https://github.com/user-attachments/assets/c7663d38-4dba-4c02-8313-3df26a2f210a" />

So in general it seems to work?

**Comment by xIrusux** [comment]
Hi @benshen-secretlab,

The behaviour you're seeing might be caused by existing templates that haven't been shared globally, or with your specific user/role. When no templates are visible to the current user, the Templates button will appear grayed out.

Saving a new template from your account should make the button available again.
To make existing templates accessible, visibility can be managed in two ways:

Share globally -> toggle the "Shared" switch when saving or editing a template
Granular access -> assign specific users or roles directly in the template settings
 
<img width="394" height="340" alt="Image" src="https://github.com/user-attachments/assets/cda6f5ea-d865-4db7-aa96-c6b30d15f6ba" />

Could you please also confirm whether you are still experiencing the second issue (Cancel button not responding in the enlarged grid config modal?

**Comment by benshen-secretlab** [comment]
Regarding Issue 1, the issue still persists. I tried saving it using both the Global and role-based (admin role) options, but the button still shows `No saved template available`.

Regarding Issue 2, the button has been renamed to `Discard all changes`, and it is functioning correctly now.

**Comment by benshen-secretlab** [comment]
Screen Recording: [pimcore issue.zip](https://github.com/user-attachments/files/28308858/pimcore.issue.zip)

**Comment by xIrusux** [comment]
Thanks for the additional info and the screen-recording. I am assuming the recording was taken while logged in as the admin user?
Unfortunately I can not reproduce this locally. Could you help narrow it down with a few checks?

1. After saving, with the button still dim, does a hard reload make the templates appear?
2. Does it reproduce in incognito / a different browser?
3. What are your versions of the pimcore/studio-backend-bundle and pimcore/studio-ui-bundle?
4. With Redux DevTools open while the button is dim: in api.queries, could you please find entries starting with dataObjectListSavedGridConfigurations. How many are there, what are their keys (they encode the classId), and does any have data.items.length > 0? A screenshot is perfect.

Thank you.

**Comment by benshen-secretlab** [comment]
Hi @xIrusux, thanks for following up.

I am logged as the admin user .

1. It didn't, the button remains dim
2. Yeah, it is reproducible under chrome incognito mode and MS edge
3. pimcore/studio-backend-bundle: `v2026.1.4` ,  pimcore/studio-ui-bundle: `v2026.1.4`
4.  2 entries found. 1 entry for pending and 1 entry for fulfilled. `data.items.length` is 0.

<img width="2536" height="960" alt="Image" src="https://github.com/user-attachments/assets/dad0fc9a-7bb6-4f19-b630-a55d98b11d2e" />

<img width="2436" height="1126" alt="Image" src="https://github.com/user-attachments/assets/6b9c5f38-43db-4761-8c00-04a127781181" />

**Comment by benshen-secretlab** [comment]
Hi @xIrusux , thanks for the hints.

I found the root cause. The issue was caused by the `classId` column in the `bundle_studio_grid_configurations` table being limited to `10` characters, which caused the `classId` value to be truncated.

Would the Pimcore Team be able to alter the table structure to support a larger key length for this case?

<img width="1014" height="260" alt="Image" src="https://github.com/user-attachments/assets/6b6d157c-328a-46e2-9a92-59477c1abf70" />

<img width="1218" height="394" alt="Image" src="https://github.com/user-attachments/assets/3cd708f5-fcfe-4ac1-8d97-86380d76bb20" />

**Comment by xIrusux** [comment]
Hi @benshen-secretlab glad to hear we found the root cause. I have transferred this issue to the backend team who will take another look at this. 
Thanks,

**Comment by lukmzig** [comment]
Hi @benshen-secretlab, thank you for reporting this issues and for the thorough analysis. The [fix](https://github.com/pimcore/studio-backend-bundle/pull/1868) and migration will be included in the next bugfix release.
